devservices/cognito-user-pools/src/main/java/io/quarkus/amazon/devservices/cognitouserpools/MotoContainer.java

```diff
@@ -29,7 +26,15 @@ public URI getEndpointOverride() {
             final String address = getHost();
             String ipAddress = address;
             // resolve IP address and use that as the endpoint so that path-style access is automatically used by client
-            ipAddress = InetAddress.getByName(address).getHostAddress();
+            final InetAddress fetchedIpAddress = InetAddress.getByName(address);
+
+            if (fetchedIpAddress instanceof Inet6Address) {
+                // RFC 2732 specifies that IPv6 addresses should be surrounded by brackets when used as a URI reference
+                ipAddress = String.format("[%s]", fetchedIpAddress.getHostAddress());
+            } else {
+                ipAddress = fetchedIpAddress.getHostAddress();
+            }
+
             return new URI("http://" + ipAddress + ":" + getMappedPort(PORT));
         } catch (UnknownHostException | URISyntaxException e) {
```

Comment:
I don't think we even need to resolve `address` in the first place. It should come from an Uri from the docker client and should already be in the appropriate form. AFAIK, cognito doesn't use path-style addressing like s3 so it's unnecessary.

Would you mind testing with just

```java
    public URI getEndpointOverride() {
        try {
            final String host = getHost();
            return new URI("http://" + host + ":" + getMappedPort(PORT));
        } catch (URISyntaxException e) {
            throw new IllegalStateException("Cannot obtain endpoint URL", e);
        }
    }
```

Unfortunately, for other services, LocalStack is doing this. For S3, it proved to be unsuccessful and we must force path-style addressing when using an IP other than localhost.

Reply:
@krizzje Do you had time to test the proposed solution ?